stan/math/prim/mat/prob/normal_id_glm_lpdf.hpp

```diff
@@ -86,56 +92,45 @@ normal_id_glm_lpdf(const T_y &y, const T_x &x, const T_alpha &alpha,
   if (!include_summand<propto, T_y, T_x, T_alpha, T_beta, T_scale>::value)
     return 0.0;
 
-  // Set up data structures and compute log-density.
-  if (include_summand<propto>::value)
-    logp += NEG_LOG_SQRT_TWO_PI * N;
-
-  scalar_seq_view<T_scale> sigma_vec(sigma);
+  const auto &x_val = value_of_rec(x);
+  const auto &beta_val = value_of_rec(beta);
+  const auto &alpha_val = value_of_rec(alpha);
+  const auto &sigma_val = value_of_rec(sigma);
+  const auto &y_val = value_of_rec(y);
 
-  Matrix<T_partials_return, Dynamic, 1> beta_dbl(M, 1);
-  {
-    scalar_seq_view<T_beta> beta_vec(beta);
-    for (size_t m = 0; m < M; ++m) {
-      beta_dbl[m] = value_of(beta_vec[m]);
-    }
-  }
+  const auto &beta_val_vec = as_column_vector_or_scalar(beta_val);
+  const auto &alpha_val_vec = as_column_vector_or_scalar(alpha_val);
+  const auto &sigma_val_vec = as_column_vector_or_scalar(sigma_val);
+  const auto &y_val_vec = as_column_vector_or_scalar(y_val);
 
-  Array<T_partials_return, Dynamic, 1> mu_minus_alpha_dbl
-      = (value_of(x) * beta_dbl).array();
-  Array<T_partials_return, Dynamic, 1> inv_sigma(N, 1);
-  scalar_seq_view<T_alpha> alpha_vec(alpha);
-  scalar_seq_view<T_y> y_vec(y);
-  Array<T_partials_return, Dynamic, 1> y_minus_mu_over_sigma(N, 1);
-  Array<T_partials_return, Dynamic, 1> y_minus_mu_over_sigma_squared(N, 1);
-  for (size_t n = 0; n < N; ++n) {
-    if (include_summand<propto, T_scale>::value)
-      logp -= log(value_of(sigma_vec[n]));
-    inv_sigma[n] = 1 / value_of(sigma_vec[n]);
-    y_minus_mu_over_sigma[n]
-        = (value_of(y_vec[n]) - mu_minus_alpha_dbl[n] - value_of(alpha_vec[n]))
-          * inv_sigma[n];
-    y_minus_mu_over_sigma_squared[n] = square(y_minus_mu_over_sigma[n]);
-  }
+  T_scale_val inv_sigma = 1 / as_array_or_scalar(sigma_val_vec);
```

Comment:
[optional]
This is another palce where our `inv(v)` will be more efficient than `1 / v` for autodiff types.